src/dataflow-types/src/types.rs

```diff
@@ -55,6 +56,73 @@ impl PeekResponse {
     }
 }
 
+impl From<&PeekResponse> for ProtoPeekResponse {
+    fn from(x: &PeekResponse) -> Self {
+        use proto_peek_response::Kind::*;
+        use proto_peek_response::*;
+        ProtoPeekResponse {
+            kind: Some(match x {
+                PeekResponse::Rows(rows) => Rows(ProtoRows {
+                    rows: rows
+                        .iter()
+                        .map(|(r, d)| ProtoRow {
+                            row: Some(r.into()),
+                            diff: usize::from(*d).into_proto(),
+                        })
+                        .collect(),
+                }),
+                PeekResponse::Error(err) => proto_peek_response::Kind::Error(err.clone()),
+                PeekResponse::Canceled => Canceled(()),
+            }),
+        }
+    }
+}
+
+impl TryFrom<ProtoPeekResponse> for PeekResponse {
+    type Error = TryFromProtoError;
+
+    fn try_from(x: ProtoPeekResponse) -> Result<Self, TryFromProtoError> {
+        use proto_peek_response::Kind::*;
+        match x.kind {
+            Some(Rows(rows)) => Ok(PeekResponse::Rows(
+                rows.rows
+                    .into_iter()
+                    .map(|row| {
+                        Ok((
+                            row.row.try_into_if_some("ProtoRow::row")?,
+                            usize::from_proto(row.diff)?.try_into()?,
```

Comment:
Can we implement `ProtoRepr` for `NonZeroUsize` (the `Proto` representation should  be the same as for `usize`)?